ignite/engine/engine.py

```diff
@@ -728,6 +713,53 @@ def _get_data_length(self, data: Iterable) -> Optional[int]:
             pass
         return None
 
+    def _check_and_set_max_epochs(self, max_epochs: Optional[int] = None) -> None:
+        if max_epochs is not None:
+            if max_epochs < 1:
+                raise ValueError("Argument max_epochs is invalid. Please, set a correct max_epochs positive value")
+            if self.state.max_epochs is not None and max_epochs <= self.state.epoch:
+                raise ValueError(
+                    "Argument max_epochs should be larger than the current epoch "
+                    f"defined in the state: {max_epochs} vs {self.state.epoch}. "
+                    "Please, set engine.state.max_epochs = None "
+                    "before calling engine.run() in order to restart the training from the beginning."
+                )
+            self.state.max_epochs = max_epochs
+
+    def _check_and_set_max_iters(self, max_iters: Optional[int] = None) -> None:
+        if max_iters is not None:
+            if max_iters < 1:
+                raise ValueError("Argument max_iters is invalid. Please, set a correct max_iters positive value")
+            if (self.state.max_iters is not None) and max_iters <= self.state.iteration:
+                raise ValueError(
+                    "Argument max_iters should be larger than the current iteration "
+                    f"defined in the state: {max_iters} vs {self.state.iteration}. "
+                    "Please, set engine.state.max_iters = None "
+                    "before calling engine.run() in order to restart the training from the beginning."
+                )
+            self.state.max_iters = max_iters
+
+    def _check_and_set_epoch_length(self, data: Iterable, epoch_length: Optional[int] = None) -> None:
+        # Can't we accept a redefinition ?
```

Comment:
I think we can't because the relationship between epoch and iteration becomes invalid then. By the way in the case `epoch_length` is changed, we could reinitialize the state (with epoch and iteration zero along with a message to the user) as well.